src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/request/TemplatePathVisitor.java

```diff
@@ -286,6 +292,37 @@ private boolean canCompleteXpath(Object extradata) {
                     || (extradata instanceof Boolean && ((Boolean) extradata).booleanValue());
         }
 
+        private String completeXpath(String source, String xpath) {
+            String sepSrc = "/";
+            String sepXpath = "/";
+            if (source.indexOf(sepSrc) == -1) sepSrc = "\\.";
+            if (xpath.indexOf(sepXpath) == -1) sepXpath = "\\.";
+
+            String[] sourceSteps = source.split(sepSrc);
+            String[] xpathSteps = xpath.split(sepXpath);
+
+            List<String> sourceList = new LinkedList<>();
+            List<String> xpathList = new LinkedList<>();
+            Collections.addAll(sourceList, sourceSteps);
+            Collections.addAll(xpathList, xpathSteps);
+
+            // does the xpath contains all or some portions of the source already?
+            List<String> intersection = new LinkedList<>(xpathList);
+            intersection.retainAll(sourceList);
```

Comment:
I'm confused... what if they share bits, but not in the same order? Recursive data structures might have stuff like "a/b/a/b", legit, but it does not mean that you can compare "a/a" with "a/b/a/b".